ts/features/bonus/bpd/components/superCashbackRanking/SuperCashbackRanking.tsx

```diff
@@ -0,0 +1,159 @@
+    ) {
+      return (
+        <>
+          <FirstPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.maxTransactionNumber}
+          />
+          <UserPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.transactionNumber}
+            userPosition={props.rankingInformation.ranking}
+          />
+        </>
+      );
+    } else if (
+      props.rankingInformation.ranking < props.selectedPeriod.minPosition &&
+      props.rankingInformation.ranking > 1
+    ) {
+      return (
+        <>
+          <FirstPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.maxTransactionNumber}
+          />
+          <UserPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.transactionNumber}
+            userPosition={props.rankingInformation.ranking}
+          />
+          <LastPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.minTransactionNumber}
+            lastAvailablePosition={props.selectedPeriod.minPosition}
+          />
+        </>
+      );
+    } else {
+      return (
+        <>
+          <UserPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.transactionNumber}
+            userPosition={props.rankingInformation.ranking}
+          />
+          <LastPositionItem
+            superCashbackAmount={props.selectedPeriod.superCashbackAmount}
+            transactionsNumber={props.rankingInformation.minTransactionNumber}
+            lastAvailablePosition={props.selectedPeriod.minPosition}
+          />
+        </>
+      );
+    }
+  }
+
+  return <View />;
+};
+
+const CSS_STYLE = `
+body {
+  color: ${IOColors.bluegreyDark}
+}
+`;
+
+const calculateEndDate = (selectedPeriod: BpdPeriodWithInfo | undefined) =>
+  fromNullable(selectedPeriod).fold("", p => {
+    const endDate = new Date(p.endDate.getTime());
+    endDate.setDate(endDate.getDate() + p.gracePeriod);
+
+    return localeDateFormat(endDate, I18n.t("global.dateFormats.shortFormat"));
+  });
+
+const SuperCashbackRanking: React.FunctionComponent<Props> = (props: Props) => (
+  <>
+    <View spacer={true} />
+    <RankingItems {...props} />
+    <View spacer={true} small />
+    <ItemSeparatorComponent noPadded={true} />
+    <View spacer={true} />
+    <H3>{I18n.t("bonus.bpd.details.superCashback.howItWorks.title")}</H3>
+    <View spacer={true} />
+    <Markdown cssStyle={CSS_STYLE}>
+      {I18n.t("bonus.bpd.details.superCashback.howItWorks.body", {
+        endDate: calculateEndDate(props.selectedPeriod)
```

Comment:
What happen if we can't calculate the date? we show a trunked information?
An improvement could return a `string | undefined` (or `Option<string>`) and render this `Markdown` only when we have a valid string?